source/dotnet/Library/AdaptiveCards/AdaptiveBaseElement.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Xml.Serialization;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace AdaptiveCards
+{
+    public class AdaptiveInternalID
+    {
+        public static uint Invalid = 0;
+        private static uint CurrentInternalID = 1;
+        private uint ID { get; }
+
+        public AdaptiveInternalID()
+        {
+            ID = Invalid;
+        }
+
+        private AdaptiveInternalID(uint id)
+        {
+            ID = id;
+        }
+
+        public static AdaptiveInternalID Current()
+        {
+            return new AdaptiveInternalID(CurrentInternalID);
+        }
+
+        public static AdaptiveInternalID Next()
+        {
+            CurrentInternalID++;
+
+            // handle overflow case
+            if (CurrentInternalID == AdaptiveInternalID.Invalid)
+            {
+                CurrentInternalID++;
+            }
+
+            return Current();
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj == null || GetType() != obj.GetType())
+            {
+                return false;
+            }
+
+            return ((AdaptiveInternalID)obj).ID == this.ID;
+        }
+
+        public override int GetHashCode()
+        {
+            return base.GetHashCode();
+        }
+
+        public override string ToString()
+        {
+            return ID.ToString();
+        }
+    }
+
+    public class AdaptiveFallbackElement
+    {
+        public static readonly string drop = "drop";
```

Comment:
>drop [](start = 38, length = 4)

Nit: this could be a const no?

Reply:
can't be `const` because it's also `static`

---
In reply to: [278318026](https://github.com/Microsoft/AdaptiveCards/pull/2560#discussion_r278318026) [](ancestors = 278318026)

Reply:
arent const's in C#  static by nature? 

---
In reply to: [278325378](https://github.com/Microsoft/AdaptiveCards/pull/2560#discussion_r278325378) [](ancestors = 278325378,278318026)